catalog/security/catalog-security-ingestplugin/src/main/java/ddf/catalog/security/ingest/IngestPlugin.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.security.ingest;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.common.util.Security;
+import ddf.security.permission.CollectionPermission;
+import ddf.security.permission.KeyValuePermission;
+
+/**
+ * IngestPlugin is a PreIngestPlugin that restricts the create/update/delete operations
+ * on the catalog to a group defined by a set of configurable user attributes.
+ */
+public class IngestPlugin implements PreIngestPlugin {
```

Comment:
This would actually be a CUD since it doesn't do anything with query (read). My understanding was that anything that modified the catalog was considered part of 'ingesting' and it's what Phill requested.